src/inmanta/deploy/state.py

```diff
@@ -183,29 +186,28 @@ def reset(self) -> None:
 
     def block_resource(self, resource: ResourceIdStr, details: ResourceDetails, transient: bool) -> None:
         """
-        Mark the given resource as blocked, i.e. it's not deployable.
+        Mark the given resource as blocked, i.e. it's not deployable. This method updates the resource details
+        and the resource_status.
 
         :param resource: The resource that should be blocked.
         :param details: The details of the resource that should be blocked.
         :param transient: True iff the given resource is blocked transitively. It's blocked because one of its dependencies
                           is blocked.
         """
         self.resources[resource] = details
-        blocked_status = BlockedStatus.TRANSIENT if transient else BlockedStatus.YES
         if resource in self.resource_state:
             if not transient:
                 self.resource_state[resource].status = ResourceStatus.UNDEFINED
-            if self.resource_state[resource].blocked is not BlockedStatus.YES:
-                self.resource_state[resource].blocked = blocked_status
+            self.resource_state[resource].blocked = BlockedStatus.YES
         else:
             resource_status = ResourceStatus.UNDEFINED if not transient else ResourceStatus.HAS_UPDATE
```

Comment:
I missed this in the previous review, but shouldn't this be `ResourceStatus.NEW` instead of `HAS_UPDATE`?